File: leetcode/1802-number-of-students-unable-to-eat-lunch/number-of-students-unable-to-eat-lunch.cpp
class Solution {
public:
    int countStudents(vector<int>& students, vector<int>& sandwiches) {
        int count0=0;
        int count1=0;
        for(int i=0;i<students.size();i++){
            if(students[i]==0){
                count0++;
            }
            else{
                count1++;
            }
        }
        for(int j=0;j<sandwiches.size();j++){
            if(sandwiches[j]){
                if(count1>0){
                count1--;

                }
                else{
                    return count0;
                }
            }
            else{
                if(count0>0){
                    count0--;
                }
                else{
                    return count1;
                }
            }
        }
        return 0;
    }
};

File: leetcode/1802-number-of-students-unable-to-eat-lunch/number-of-students-unable-to-eat-lunch.py
class Solution(object):
    def countStudents(self, students, sandwiches):
        """
        :type students: List[int]
        :type sandwiches: List[int]
        :rtype: int
        """
        numOnes = 0
        numZeros = 0
        for i in students:
            if i == 0:
                numZeros += 1
            else:
                numOnes += 1
        for j in sandwiches:
            if j == 1:
                if numOnes > 0:
                    numOnes -= 1
                else:
                    return numOnes+numZeros
            else:
                if numZeros > 0:
                    numZeros -= 1
                else:
                    return numOnes+numZeros
        return 0

        